Bottom: 6b17969e24feee29f67db8ca49834403c8d7f90c
Top:    deac90fd7be1c88ecf1ac2f403cc66dd5a37be36
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-17 01:59:17 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 30f6623..40cfde9 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -309,6 +309,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
